src/System.Net.Security/tests/FunctionalTests/SslStreamAlpnTests.cs

```diff
@@ -180,6 +189,38 @@ public async Task SslStream_StreamToStream_Alpn_NonMatchingProtocols_Fail()
             }
         }
 
+        [OuterLoop("Uses external server")]
+        [PlatformSpecific(TestPlatforms.OSX)]
+        [Theory]
+        [MemberData(nameof(Http2Servers))]
+        public async Task SslStream_OSX_Alpn_Success(Uri server)
+        {
+            using (TcpClient client = new TcpClient())
+            {
+                try
+                {
+                    await client.ConnectAsync(server.Host, server.Port);
+                    using (SslStream clientStream = new SslStream(client.GetStream(), leaveInnerStreamOpen: false))
+                    {
+                        SslClientAuthenticationOptions clientOptions = new SslClientAuthenticationOptions
+                        {
+                            ApplicationProtocols = new List<SslApplicationProtocol> { SslApplicationProtocol.Http2 ,  SslApplicationProtocol.Http11 },
+                            TargetHost =  server.Host
+                        };
+
+                        await clientStream.AuthenticateAsClientAsync(clientOptions, CancellationToken.None);
+                        Assert.Equal("h2", clientStream.NegotiatedApplicationProtocol.ToString());
+                    }
+                }
+                catch (Exception e)
+                {
+                    // Failures to connect do not cause test failure.
+                    _output.WriteLine("Unable to connect: {0}", e);
```

Comment:
I think writing to stdout works correctly (at least worked for me last time on windows and linux), having the exception directly tells you the exact place and makes it slightly easier to debug (not sure if helps in async scenario as in here)

Reply:
Writing to xunit handle allows to preserve it in test result database. (as it gets written to result xml and processed by CI)
I used stdout only for local debug and there are already other tests using this pattern.

Reply:
Yes, we should not be using stdout in Xunit.  We should only be using the `ITestOutputHelper` pattern.